File: console-sample/src/main/java/com/jraska/console/sample/ChattyAsync.java
package com.jraska.console.sample;

import android.os.AsyncTask;
import com.jraska.console.Console;
import timber.log.Timber;

public class ChattyAsync extends AsyncTask<Long, Void, Void> {
  @Override protected Void doInBackground(Long... params) {
    long time = params[0];

    Console.writeLine("Async just started");
    sleep(time);
    Timber.d("Message from async after %s ms", time);
    sleep(time);
    Timber.w("Note done yet, %s ms left", time);
    sleep(time);
    Timber.i("Finally shutting up");

    return null;
  }

  private void sleep(long time) {
    try {
      Thread.sleep(time);
    }
    catch (InterruptedException e) {
      throw new RuntimeException(e);
    }
  }
}

File: console-sample/src/main/java/com/jraska/console/sample/ChattyAsync.kt
package com.jraska.console.sample

import android.os.AsyncTask
import com.jraska.console.Console
import timber.log.Timber
import java.lang.Thread.sleep

class ChattyAsync(val sleepTime: Long) : AsyncTask<Any, Void, Void>() {
  override fun doInBackground(vararg params: Any?): Void? {

    Console.writeLine("Async just started")
    sleep(sleepTime)
    Timber.d("Message from async after %s ms", sleepTime)
    sleep(sleepTime)
    Timber.w("Note done yet, %s ms left", sleepTime)
    sleep(sleepTime)
    Timber.i("Finally shutting up")

    return null
  }
}
